rs/universal_canister/lib/src/lib.rs

```diff
@@ -7,17 +7,15 @@
 
 pub mod management;
 
-use hex_literal::hex;
 use ic_types::Cycles;
 use universal_canister::Ops;
-/// The binary of the universal canister as compiled from
-/// `rs/universal_canister/impl`.
-///
-/// For steps on how to produce it, please see the README in
-/// `rs/universal_canister`.
-pub const UNIVERSAL_CANISTER_WASM: &[u8] = include_bytes!("universal-canister.wasm.gz");
-pub const UNIVERSAL_CANISTER_WASM_SHA256: [u8; 32] =
-    hex!("9c0b4ed1d729ffdd0e8d194df3be621f58a2fb86e1f1bbf556d89f43637de303");
+
+/// The binary of the universal canister as compiled from `rs/universal_canister/impl`.
+pub const UNIVERSAL_CANISTER_WASM: &[u8] = include_bytes!(env!("UNIVERSAL_CANISTER_WASM_PATH"));
+
+pub fn get_universal_canister_wasm_sha256() -> [u8; 32] {
+    ic_crypto_sha2::Sha256::hash(UNIVERSAL_CANISTER_WASM)
+}
```

Comment:
if you are using a `build.rs` anyway, how about computing the sha from there anyway?

Reply:
I'm not sure how to get the resulting hash into the source file as a const [u8; 32]. Would you know how to do that?

Reply:
Also I'm not sure making this more complicated is worth it. These hashes are not used in hot loops so performance shouldn't be a concern.

Reply:
I simplified it further and now only use `build.rs` for the cargo invocations and not for the bazel build anymore.